### DISCUSSION TITLE
--lxde-config

### DISCUSSION THREAD

**Discussion by dshaw619** [gitbls/sdm#44] [Q&A] [answered]
I am trying to configure lxterminal to my liking when customizing an image.  Per https://github.com/gitbls/sdm/wiki/Using-LXDE-Config I added an --lxde-config switch to my --customize command and copied a working lxterminal.conf from a working Bullseye system.  However, I repeatedly get an error message saying the .conf file can't be found.

```
doug@dv6-Notebook-PC:~$ sudo /usr/local/sdm/sdm --customize \
--wpa ~/Documents/Briefcase/It/Raspberry_Pi/Setup/11-Bullseye/wpa_supplicant.conf \
--keymap us \
--locale en_US.UTF-8 \
--timezone America/Los_Angeles \
--wifi-country US \
--disable piwiz \
--svc-disable userconfig \
--bootset serial=0,boot_behaviour=B4,boot_wait=0 \
--lxde-config lxterminal:~/Documents/Briefcase/It/Raspberry_Pi/Setup/11-Bullseye/lxterminal.conf \
--restart \
--hostname raspberrypi \
--user des \
--extend --xmb 1024 \
~/Downloads/bullseye-armhf-customized.img
[sudo] password for doug:             
? --lxde-config lxterminalconf config file '~/Documents/Briefcase/It/Raspberry_Pi/Setup/11-Bullseye/lxterminal.conf' not found
```
But the file is actually there ...

```
doug@dv6-Notebook-PC:~$ ls -l ~/Documents/Briefcase/It/Raspberry_Pi/Setup/11-Bullseye
total 80
-rw------- 1 doug doug  175 Oct  4  2020 cmdline.txt
-rw------- 1 doug doug 1788 Jun 25  2019 config.txt
-rw------- 1 doug doug 1450 Nov 22 23:01 lxterminal.conf
-rwxrwxr-x 1 doug doug  312 Nov 22 19:12 sdm-burn.sh
-rwxrwxr-x 1 doug doug  905 Nov 23 13:22 sdm-customize.sh
-rw------- 1 doug doug    0 Aug 28  2018 ssh
-rw------- 1 doug doug  379 Oct  5 15:26 wpa_supplicant.conf
```
I've tried with the .conf file in different locations (e.g. shorter pathnames) and both with the --customize command in a script and directly from the command line as above.  I found where the message is generated at line 959 in sdm, but am not proficient enough with bash to figure out what might be occurring or what I might be doing wrong.

**Comment by gitbls** [comment]
I'm not in a place where I can have a look at the code, but I'm pretty sure that using "~" to represent your home directory doesn't work. Use the /full/path to the file and see if that works as it should.

**Reply by dshaw619** [reply]
I tried sdm --customize from the command line with the full path (/home/doug/...) and it worked.  Then I made the same edit to the script which I'm using to copy the initial .img into one I can customize and then run sdm; that failed.  For good measure, I copied the sdm command from that script and executed it directly on the command line; it again failed.

```
doug@dv6-Notebook-PC:~$ sudo /usr/local/sdm/sdm --customize \
--wpa ~/Documents/Briefcase/It/Raspberry_Pi/Setup/11-Bullseye/wpa_supplicant.conf \
--keymap us \
--locale en_US.UTF-8 \
--timezone America/Los_Angeles \
--wifi-country US \
--disable piwiz \
--svc-disable userconfig \
--bootset serial=0,boot_behaviour=B4,boot_wait=0 \
--lxde-config lxterminal:home/doug/Documents/Briefcase/It/Raspberry_Pi/Setup/11-Bullseye/lxterminal.conf \
--restart \
--hostname raspberrypi \
--user des \
--extend --xmb 1024 \
~/Downloads/bullseye-armhf-customized.img
? --lxde-config lxterminalconf config file 'home/doug/Documents/Briefcase/It/Raspberry_Pi/Setup/11-Bullseye/lxterminal.conf' not found
```

**Reply by dshaw619** [reply]
Oops ... should have been /home/doug/...  Corrected my error and retrying now.

**Reply by dshaw619** [reply]
Worked fine both from the command line and in my script when I (correctly) specified .conf file as /home/doug/... (facepalm!)
Any short explanation of why this is necessary and ~/... doesn't work?  vs. --wpa which works with ~/...?  I'm guessing it might have to do with parsing lxterminal:<filename.conf>.

**Comment by gitbls** [comment]
Good question! The lx parsing code needs a closer look, which I'll do when I'm back at my home workstation.

**Comment by gitbls** [comment, marked as answer]
The difference between the processing for `--wpa` and `--lxde-config` with respect to "~" is that bash itself expand the switch value with `--wpa`. The `--lxde-config` switch value is complex, and bash doesn't know how to process it before starting sdm, as it does with `--wpa`.

The fix is fairly simple, except for the process of ensuring that all instances where it needs to be fixed...get fixed.

I'd suggest that you use $HOME instead of "~", which bash will correctly expand.

**Reply by dshaw619** [reply]
Will do.  Thanks.
